debezium-connector-cassandra/src/main/java/io/debezium/connector/cassandra/SchemaProcessor.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.cassandra;
+
+import java.time.Duration;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * The schema processor is responsible for periodically
```

Comment:
Exactly. This is one caveat of the Cassandra commit logs. It's actually even more nuanced than that: the mutations in commit log are not always in order, so we could process an event at time `T` followed by another event at time `T-1`. This means the concept of a `DatabaseHistory` is very difficult to implement here since we can't rely on file position to determine order of events, instead we must rely on (client) timestamp. But since commit logs don't include schema change, we won't be have the timestamp for schema change...
Haven't been able to come up with a great solution on this.